### DISCUSSION TITLE
What does KP stand for; context within the policy for limiations?

### DISCUSSION THREAD

**Discussion by Exaskryz** [ImageMagick/ImageMagick#6751] [Help] [answered]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Ubuntu 22.04 LTS

### Description

>   <policy domain="resource" name="memory" value="256MiB"/>
>   <policy domain="resource" name="map" value="512MiB"/>
>   <policy domain="resource" name="width" value="16KP"/>
>   <policy domain="resource" name="height" value="16KP"/>

What do those 16KP mean?

This is part of my policy and I figure I should be expanding the width and height values because trying to append a few images together results in it breaking. 

The command is `convert "$@" -append ImageMagickOutput/collage.jpg` and I have a list of input files when I call it via terminal. They're all the same size images, 1206x600. It will yell at me telling me 

> convert-im6.q16: width or height exceeds limit `ImageMagickOutput/Legend-PvP.png' @ error/cache.c/OpenPixelCache/3909.

Legend-PvP being the first image in the list of arguments.

And then it outputs all the other input files under the pseudoarray collage-0, collage-1, etc.

This worked when the image files were smaller, not a problem. So I figured I just need to increase the width and height policies there to 99KP. And doing so works, my command doesn't throw an error for exceeding width/height limits.

My question is, what does KP stand for? Kilopixel? Nothing on duckduckgo nor the documentation at https://imagemagick.org/script/security-policy.php seems to say what a KP is so I know I'm setting appropriate thresholds. Any clarification would be appreciated!

**Comment by snibgo** [comment, marked as answer]
In that context, "K" is thousand (1000), and "P" is pixels.

An alternative is "Ki", meaning "binary thousand", ie 1024.

"B" is bytes. "M" is mega, "P" is peta, etc. (Yes, that is a bit naff, so "PP" would be "peta pixels".)

**Comment by urban-warrior** [comment]
See https://www.nist.gov/pml/owm/metric-si-prefixes.
